### PR TITLE
Handle ungood build.gradle parsing results better

### PR DESCRIPTION
**Pull request by jsonperl**
Seems that the gradle parsing service does a poor job when there's no version requested.

### CODE REVIEW

File: lib/bibliothecary/parsers/maven.rb [hunk 1]
```diff
@@ -205,8 +205,8 @@ def self.parse_gradle(manifest)
         json = JSON.parse(response.body)
         return [] unless json['dependencies']
         json['dependencies'].map do |dependency|
-          name = [dependency["group"], dependency["name"]].join(':')
-          next unless name =~ (/[A-Za-z0-9_-]+\.[A-Za-z0-9_-]+(\.[A-Za-z0-9_-])?\:[A-Za-z0-9_-]/)
+          name = gradle_dependency_name(dependency["group"], dependency["name"])
+          next unless name =~ /[\w-]+\.[\w_-]+(\.[\w-])?\:[\w-]/
```
Comment (tiegz) [inline lib/bibliothecary/parsers/maven.rb:209]: 👍

File: spec/parsers/maven_spec.rb [hunk 2]
```diff
@@ -166,24 +166,46 @@
...
+          testCompile "hello.there:im.a.dep:$versionThing" // I am a comment
+        }
+      FILE
+
+      expect(described_class.analyse_contents('build.gradle', source)[:dependencies]).to match_array([
+        {:name=>"com.whatever:liblib", :requirement=>"1.2.3", :type=>"compile"},
+        {:name=>"this.is.a.group:greatlib", :requirement=>"", :type=>"compile"},
+        {:name=>"com.fasterxml.jackson.core:jackson-databind", :requirement=>"", :type=>"compile"},
+        {:name=>"this.thing:neat", :requirement=>"", :type=>"compileOnly"},
+        {:name=>"hello.there:im.a.dep:$versionThing", :requirement=>"", :type=>"testCompile"},
```
Comment (tiegz) [inline spec/parsers/maven_spec.rb:186]: We don't want to capture the version when it's not a number?

Comment (jsonperl) [reply spec/parsers/maven_spec.rb:186]: Hmm good question, lemme poke a little.

Comment (jsonperl) [reply spec/parsers/maven_spec.rb:186]: Yea honestly, we don't. I was going to hack it in somehow, but it's just bad data.

**Review by mellisio — APPROVED** [review]

**Review by tiegz — APPROVED** [review]

**Pull request merged**